protocol/txntype.go

```diff
@@ -20,8 +20,11 @@ package protocol
 // in a block.  They are used in the data/transaction package and the REST API.
 
 // TxType is the type of the transaction written to the ledger
+//msgp:allocbound TxType txTypeMaxLen
 type TxType string
 
+const txTypeMaxLen = 7
```

Comment:
add a test checking all txTypeMaxLen against all possible TxType constants with AST

Reply:
I don't think that we need to do this since this would break only if we added a new txtype with longer name. If we did that we would be writing tests of making sure the tests work properly and none of those would pass since msgp would fail to unmarshall messages exceeding this allocbound